python/imageutils.py

```diff
@@ -295,125 +185,55 @@ def reconstruct_block_slice(self, reconstructed, legend, mem):
 
         print y_size, z_size, x_size
 
+        total_read = 0
+        total_seek = 0
+        total_defrag = 0
+        total_write = 0
+
         with open(legend, "r") as legend:
 
             remaining_mem = mem
-            data = None
-            start_x = 0
-            start_z = 0
-            start_y = 0
+            data_dict = {}
 
-            prev_block_start = (0,0,0) 
-            for block_name in legend:
-                
-                block_name = block_name.strip()
-                print "Reading block {0}".format(block_name)
-                block_pos = split_ext(block_name)[0].split('_')
-
-                #we will start our (partial) slice at these block coordinates
-                if remaining_mem == mem:
-                    start_x = int(block_pos[-1])
-                    start_z = int(block_pos[-2])
-                    start_y = int(block_pos[-3])
-                    
-                block = nib.load(block_name)
-                block_shape = block.header.get_data_shape()
-                bytes_per_voxel_block = block.header['bitpix']/8
-                block_bytes = block.header.single_vox_offset + bytes_per_voxel_block*(block_shape[0]*block_shape[1]*block_shape[2])
+            unread_split = None
+
+            for split_name in legend:
+               
+                print "Remaining mem: ", remaining_mem 
+                split_name = split_name.strip()
+
+                if unread_split is not None:
+                    unread_split, remaining_mem, read_time = insert_in_dict(data_dict, unread_split, remaining_mem, bytes_per_voxel, y_size, z_size)
+
+                if unread_split is None:    
```

Comment:
To me, an `unread_split` is just supposed to be one that was not able to be read during the previous clustered_reading, and therefore, needs to be read during the current one. There are two insert_in_dict statements because the splits are being read in a for loop and I want to make sure i wrote the "missed" block and write the current one.